rbc/external.py

```diff
@@ -0,0 +1,81 @@
+# This code has heavily inspired in the numba.extending.intrisic code
+
+from numba.core import extending, funcdesc, types
+from rbc.typesystem import Type
+
+
+class External:
+    @classmethod
+    def fromobject(cls, signature, name=None):
+        # Make inner function for the actual work
+        t = Type.fromobject(signature)
+        if not t.is_function:
+            raise ValueError("signature must represent a function type")
+
+        if name is None:
+            name = t.name
+        if not name:
+            raise ValueError(
+                f"external function name not specified for signature {signature}"
+            )
+
+        return cls(name, t.tonumba())
+
+    def __init__(self, name: str, signature: types.FunctionType):
+        """
+        Parameters
+        ----------
+        name : str
+            The name of the external function
+        signature : Numba function signature
+            A numba function type signature. i.e. (float64, float64) -> int64
+        """
+        self._signature = signature
+        self.name = name
+        self.register()
+
+    def register(self):
+        # typing
+        from numba.core.typing.templates import (
+            make_concrete_template,
+            infer_global,
+            infer,
+        )
+
+        template = make_concrete_template(
+            self.name, key=self.name, signatures=[self.signature]
```

Comment:
What happens when the `(self.name, self.signature)` pair has already registered? Would it be possible to issue a warning?

Reply:
Former will be replaced by the latter